ols/app/models/config.py

```diff
@@ -220,16 +240,31 @@ def __init__(self, data: Optional[dict] = None):
         super().__init__()
         if data is None:
             return
-        self.max_entries = data.get(
-            "max_entries", constants.IN_MEMORY_CACHE_MAX_ENTRIES
-        )
+
+        try:
+            self.max_entries = int(
+                data.get("max_entries", constants.IN_MEMORY_CACHE_MAX_ENTRIES)
+            )
+            if self.max_entries < 0:
+                raise ValueError
+        except ValueError:
+            raise InvalidConfigurationError("invalid max_entries for memory cache")
 
     def __eq__(self, other):
         """Compare two objects for equality."""
         if isinstance(other, MemoryConfig):
             return self.max_entries == other.max_entries
         return False
 
+    def validate_yaml(self) -> None:
+        """Validate memory cache config."""
+        if (
+            self.max_entries is None
+            or not isinstance(self.max_entries, int)
```

Comment:
is it needed? I mean, there are other "integer" config options, but these are just converted in `__init__` w/o further type checking